crates/accelerate/src/gate_direction.rs

```diff
@@ -0,0 +1,148 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2024
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use crate::nlayout::PhysicalQubit;
+use crate::target_transpiler::Target;
+use hashbrown::HashSet;
+use pyo3::prelude::*;
+use qiskit_circuit::imports;
+use qiskit_circuit::operations::OperationRef;
+use qiskit_circuit::{
+    dag_circuit::{DAGCircuit, NodeType},
+    operations::Operation,
+    packed_instruction::PackedInstruction,
+    BitType, Qubit,
+};
+use smallvec::smallvec;
+
+/// Check if the two-qubit gates follow the right direction with respect to the coupling map.
+///
+/// Args:
+///     dag: the DAGCircuit to analyze
+///
+///     coupling_edges: set of edge pairs representing a directed coupling map, against which gate directionality is checked
+///
+/// Returns:
+///     true iff all two-qubit gates comply with the coupling constraints
+#[pyfunction]
+#[pyo3(name = "check_gate_direction_coupling")]
+fn py_check_with_coupling_map(
+    py: Python,
+    dag: &DAGCircuit,
+    coupling_edges: HashSet<(Qubit, Qubit)>,
+) -> PyResult<bool> {
+    let coupling_map_check = |_: &PackedInstruction, op_args: &[Qubit]| -> bool {
+        coupling_edges.contains(&(op_args[0], op_args[1]))
```

Comment:
If you change the input type here to a fixed size array you can get away without having to build the tuple. I don't think functionally it'll make a difference but it's more concise.

```suggestion
    coupling_edges: HashSet<[Qubit; 2]>,
) -> PyResult<bool> {
    let coupling_map_check = |_: &PackedInstruction, op_args: &[Qubit]| -> bool {
        coupling_edges.contains(op_args)
```